src/TiledArray/error.h

```diff
@@ -22,43 +22,56 @@
 
 #include <TiledArray/config.h>
 
-// Check for default error checking method, which is determined by
-// TA_DEFAULT_ERROR, which is defined in TiledArray/config.h.
-#ifdef TA_DEFAULT_ERROR
-#if !defined(TA_EXCEPTION_ERROR) && !defined(TA_ASSERT_ERROR) && \
-    !defined(TA_NO_ERROR) && !defined(TA_ABORT_ERROR)
-#if TA_DEFAULT_ERROR == 0
-#define TA_NO_ERROR
-#elif TA_DEFAULT_ERROR == 1
-#define TA_EXCEPTION_ERROR
-#elif TA_DEFAULT_ERROR == 2
-#define TA_ASSERT_ERROR
-#elif TA_DEFAULT_ERROR == 3
-#define TA_ABORT_ERROR
-#endif  // TA_DEFAULT_ERROR == ?
-#endif  // !defined(TA_EXCEPTION_ERROR) && !defined(TA_ASSERT_ERROR) &&
-        // !defined(TA_NO_ERROR) && !defined(TA_ABORT_ERROR)
-#endif  // TA_DEFAULT_ERROR
-
-#include <exception>
+#ifndef TA_ASSERT_POLICY
+#define TA_ASSERT_POLICY TA_ASSERT_THROW
+#endif
+
+#define TA_STRINGIZE(s) #s
+
+#define TA_ASSERT_MESSAGE(EXPR, ...)            \
+  __FILE__ ":" TA_STRINGIZE(__LINE__) ": "      \
+  "TA_ASSERT failed: " TA_STRINGIZE(EXPR)
+
+#if TA_ASSERT_POLICY == TA_ASSERT_IGNORE
+#define TA_ASSERT(...) do { } while(0)
+#else
+#define TA_ASSERT(EXPR, ...)                                            \
+  do {                                                                  \
+    if (!(EXPR))                                                        \
+      TiledArray::assert_failed(TA_ASSERT_MESSAGE(EXPR, __VA_ARGS__));  \
+  } while(0)
+
+#endif
+
+#include <stdexcept>
+#include <string>
+
 namespace TiledArray {
 
-class Exception : public std::exception {
- public:
-  Exception(const char* m) : message_(m) {}
+void Abort();
```

Comment:
uppercase is weird for function names ... is this is to avoid clashing with std::abort? How about renaming to ta_abort() then?